edx_proctoring/api.py

```diff
@@ -1526,6 +1526,23 @@ def _calculate_allowed_mins(due_datetime, allowed_mins):
     return actual_allowed_mins, is_exam_past_due_date
 
 
+def _is_self_paced_course(user_id, course_id):
```

Comment:
@douglashall, that's what I have discovered. The `due_date` field passed within `context` is a field overriden by self-paced field override at https://github.com/edx/edx-platform/blob/master/lms/djangoapps/courseware/self_paced_overrides.py#L10 
The override gets applied at the module_render time at https://github.com/edx/edx-platform/blob/master/lms/djangoapps/courseware/module_render.py#L573
As you can see in the `self_paced_overrides`, the `due` and `start` date of all course elements are overriden. The reason for this override is precisely because we want to allow user to switch back and forth from self-paced to instructor-paced quickly.